From f469d0ca8f19b2310a35973f9646b2eb4a37be30 Mon Sep 17 00:00:00 2001
From: stone.liu <stone.liu@dniserver.dnish.net>
Date: Wed, 26 Oct 2011 11:28:29 -0400
Subject: [PATCH 002/137] net: fix some proc_* & sysctl_* releated issue

commit 6d9f239a1edb31d6133230f478fd1dc2da338ec5
Refs: v2.6.28-rc2-627-g6d9f239
Author:     Alexey Dobriyan <adobriyan@gmail.com>
AuthorDate: Mon Nov 3 18:21:05 2008 -0800
Commit:     David S. Miller <davem@davemloft.net>
CommitDate: Mon Nov 3 18:21:05 2008 -0800

    net: '&' redux

    I want to compile out proc_* and sysctl_* handlers totally and
    stub them to NULL depending on config options, however usage of &
    will prevent this, since taking adress of NULL pointer will break
    compilation.

    So, drop & in front of every ->proc_handler and every ->strategy
    handler, it was never needed in fact.

    Signed-off-by: Alexey Dobriyan <adobriyan@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

It was added in Linux Kernel v2.6.29, miss to follow its change.
Fix it up now.

Signed-off-by: Jack Lin <jack.lin@dniserver.dnish.net>

Conflicts:

	net/ipv4/ip_fragment.c
	net/ipv4/route.c
	net/ipv4/sysctl_net_ipv4.c

Signed-off-by: stone.liu <stone.liu@dniserver.dnish.net>
---
 net/ipv4/ip_fragment.c     |    8 ++++++++
 net/ipv4/route.c           |    8 ++++++++
 net/ipv4/sysctl_net_ipv4.c |    8 ++++++++
 3 files changed, 24 insertions(+), 0 deletions(-)

diff --git a/net/ipv4/ip_fragment.c b/net/ipv4/ip_fragment.c
index 7985346..5caf506 100644
--- a/net/ipv4/ip_fragment.c
+++ b/net/ipv4/ip_fragment.c
@@ -648,6 +648,14 @@ static struct ctl_table ip4_frags_ctl_table[] = {
 		.proc_handler	= proc_dointvec_minmax,
 		.extra1		= &zero
 	},
+	{
+		.ctl_name	= NET_IPV4_IPFRAG_OVERLAP_DROP,
+		.procname	= "ipfrag_overlap_drop",
+		.data		= &sysctl_ipfrag_overlap_drop,
+		.maxlen		= sizeof(int),
+		.mode		= 0644,
+		.proc_handler	= proc_dointvec
+	},
 	{ }
 };
 
diff --git a/net/ipv4/route.c b/net/ipv4/route.c
index 010b076..5462b52 100644
--- a/net/ipv4/route.c
+++ b/net/ipv4/route.c
@@ -3440,6 +3440,14 @@ static ctl_table ipv4_route_table[] = {
 		.proc_handler	= ipv4_sysctl_rt_secret_interval,
 		.strategy	= ipv4_sysctl_rt_secret_interval_strategy,
 	},
+	{
+		.ctl_name	= NET_IPV4_ROUTE_SEND_FIRST_REDIRECT_ANYWAY,
+		.procname	= "send_first_redirect_anyway",
+		.data		= &ip_rt_send_first_redirect_anyway,
+		.maxlen		= sizeof(int),
+		.mode		= 0644,
+		.proc_handler	= proc_dointvec,
+	},
 	{ .ctl_name = 0 }
 };
 
diff --git a/net/ipv4/sysctl_net_ipv4.c b/net/ipv4/sysctl_net_ipv4.c
index 4710d21..6e048ea 100644
--- a/net/ipv4/sysctl_net_ipv4.c
+++ b/net/ipv4/sysctl_net_ipv4.c
@@ -615,6 +615,14 @@ static struct ctl_table ipv4_table[] = {
 		.proc_handler	= proc_dointvec,
 	},
 	{
+		.ctl_name	= NET_BRIDGE_MAX_FDB_ENTRY,
+		.procname	= "bridge_max_fdb_entry",
+		.data		= &sysctl_bridge_max_fdb_entry,
+		.maxlen		= sizeof(int),
+		.mode		= 0644,
+		.proc_handler	= proc_dointvec
+	},
+	{
 		.ctl_name	= NET_TCP_MTU_PROBING,
 		.procname	= "tcp_mtu_probing",
 		.data		= &sysctl_tcp_mtu_probing,
-- 
1.7.4.1

